pkg/liqo-controller-manager/namespaceOffloading-controller/manage_liqo_scheduling_label.go

```diff
@@ -0,0 +1,51 @@
+package namespaceoffloadingctrl
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
+)
+
+func addLiqoSchedulingLabel(ctx context.Context, c client.Client, namespaceName string) error {
+	namespace := &corev1.Namespace{}
+	if err := c.Get(ctx, types.NamespacedName{Name: namespaceName}, namespace); err != nil {
+		klog.Errorf("%s --> Unable to get the namespace '%s'", err, namespaceName)
+		return err
+	}
+
+	if value, ok := namespace.Labels[liqoconst.SchedulingLiqoLabel]; !ok || value != liqoconst.SchedulingLiqoLabelValue {
+		if namespace.Labels == nil {
+			namespace.Labels = map[string]string{}
+		}
```

Comment:
Would move this before the previous if

Reply:
Sorry but I have not understood the reason; check a value in a nil map is a safe operation if this is the reason

Reply:
It was just a bit strange to me checking for the existence of the value in a possibly nil map and then constructing it if empty. I have to think whether the operation is safe or not, while the opposite would leave no doubts (with no performance difference). Still, just a matter of personal opinion, feel free to ignore

Reply:
ok I got the point thank you.